osh/glob_.py

```diff
@@ -446,28 +449,90 @@ def GlobToERE(pat):
 
 class Globber(object):
 
-    def __init__(self, exec_opts):
-        # type: (optview.Exec) -> None
+    def __init__(self, exec_opts, mem):
+        # type: (optview.Exec, state.Mem) -> None
         self.exec_opts = exec_opts
+        self.mem = mem
 
         # Other unimplemented bash options:
         #
-        # dotglob           dotfiles are matched
         # globstar          ** for directories
         # globasciiranges   ascii or unicode char classes (unicode by default)
         # nocaseglob
         # extglob          the @() !() syntax -- libc helps us with fnmatch(), but
         #                  not glob().
-        #
-        # NOTE: Bash also respects the GLOBIGNORE variable, but no other shells
-        # do.  Could a default GLOBIGNORE to ignore flags on the file system be
-        # part of the security solution?  It doesn't seem totally sound.
+
+    def _GetGlobIgnorePatterns(self):
+        # type: () -> Optional[List[str]]
+        """Get GLOBIGNORE patterns as a list, or None if not set."""
+        from _devbuild.gen.value_asdl import value_e
+        from _devbuild.gen.runtime_asdl import scope_e
+        
+        cell = self.mem.GetCell('GLOBIGNORE', scope_e.GlobalOnly)
+        if cell is None:
+            return None
+        
+        val = cell.val
+        if val.tag() != value_e.Str:
+            return None
+        
+        globignore = cast(value.Str, val).s  # type: str
+        if not globignore:  # Empty string
+            return None
+        
+        # Split by colon to get individual patterns, but don't split colons
+        # inside bracket expressions like [[:alnum:]]
+        patterns = []  # type: List[str]
+        current = []  # type: List[str]
+        in_bracket = False
+        
+        for i, c in enumerate(globignore):
+            if c == '[':
+                in_bracket = True
+                current.append(c)
+            elif c == ']':
+                in_bracket = False
+                current.append(c)
+            elif c == ':' and not in_bracket:
+                # Colon outside brackets is a separator
+                if current:
+                    patterns.append(''.join(current))
+                    current = []
```

Comment:
I think del current[:] produces less garbage, and I think mycpp supports it

We learned the hard way that idiomatic Python creates a lot of garbage!